woocommerce/woocommerce
# Fix double scrollbars on product editor page

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR fixes the double scrollbars in the page.  In order to do this, this PR does a few things:

* Refactors the `Layout` component to a functional component (to allow easier use of hooks)
* Creates a hook that helps create a slug from a route path and adds this class to the body to allow styling various elements on a page
* Styles the interface skeleton without fixing its position to the screen

Closes #38149 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
2. Under Features tab make sure to enable product-block-editor
3. Then visit `/wp-admin/admin.php?page=wc-admin&path=/add-product`
4. Reduce your viewport/browser height so you can scroll the page
5. Make sure that only one scrollbar is shown when scrolling the product editor
6. Narrow your viewport and make sure the header stays the correct width and location on various screen sizes (note that padding of the editor on mobile devices will be adjusted in a separate issue)
7. Open your console
8. Run `localStorage.setItem( 'debug', 'wc-admin:tracks' );` in the console.
9. Navigate to various WooCommerce Admin pages (not legacy pages), such as Analytics pages, Home, or the new Product editing experience
10. Make sure that `wcadmin_page_view` event is correctly triggered on each page
